pgmpy/base/AncestralBase.py

```diff
@@ -0,0 +1,486 @@
+from collections import deque
+from typing import Hashable, Iterable, Optional
+
+import networkx as nx
+import numpy as np
+
+
+class AncestralBase(nx.Graph):
+    def __init__(
+        self,
+        ebunch: Optional[Iterable[tuple[Hashable, Hashable]]] = None,
+        latents: set[Hashable] = set(),
+    ):
+        """
+        Ancestral graph base class.
+
+        Parameters
+        ----------
+        ebunch : Iterable[tuple], optional
+            An iterable of edges of the form (u, v, u_mark, v_mark) used to
+            initialize the graph. Each mark must be one of {">", "-", "o"}.
+            Default is None, which initializes an empty graph.
+
+        latents : set, optional
+            Set of latent (unobserved) variables in the graph. Default is
+            an empty set.
+
+        Edge Representation
```

Comment:
You can't have a separate section. I don't think it will get parsed. The content of this section needs to be at the top.